apis/r/tests/testthat/test-SeuratIngest.R

```diff
@@ -137,6 +137,96 @@ test_that("Write Assay mechanics", {
   gc()
 })
 
+test_that("Write v5 in-memory Assay mechanics", {
+  skip_if(!extended_tests())
+  skip_if_not_installed("SeuratObject", minimum_version = "5.0.2")
+
+  uri <- tempfile(pattern = "write-v5-assay")
+  collection <- SOMACollectionCreate(uri)
+  on.exit(collection$close(), add = TRUE, after = FALSE)
+
+  rna <- get_data("pbmc_small", package = "SeuratObject")[["RNA"]]
+  rna <- as(rna, "Assay5")
+  expect_no_condition(ms <- write_soma(rna, soma_parent = collection))
+  expect_s3_class(ms, 'SOMAMeasurement')
+  expect_true(ms$exists())
+  on.exit(ms$close(), add = TRUE, after = FALSE)
+
+  expect_identical(ms$uri, file.path(collection$uri, "rna"))
+  assay_hint <- .assay_version_hint("v5")
+  expect_equivalent(
+    ms$get_metadata(names(assay_hint)),
+    assay_hint[[1L]]
+  )
+  expect_identical(ms$names(), c("X", "var"))
+  expect_s3_class(ms$var, "SOMADataFrame")
+  expect_identical(setdiff(ms$var$attrnames(), "var_id"), names(rna[[]]))
+  expect_s3_class(ms$X, "SOMACollection")
+  expect_identical(ms$X$names(), SeuratObject::Layers(rna))
+  fmat <- methods::slot(rna, name = "features")
```

Comment:
Same as above with spelling out `fmat`, `cmat`, `lyr` -- just a few keystrokes for you -- single-time spend of those seconds! -- with added clarity for every reader forever after